ObjectMsg[] objects
TagMsg[] tags
ContainingMsg[] containingMap
